### PR TITLE
Added basic init script

### PR DESCRIPTION
**Pull request by maus-**
Tested successfully on CentOS. Assumes osquery config will be located in /etc/osquery/osquery.conf

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the admins verify this patch? Reply with "ok to test" to kick off a build...

File: tools/deployment/osquery.initd [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+config=/etc/osquery/osquery.conf
+lockfile=/var/lock/subsys/osqueryd
+pidfile=/var/run/osquery.pid
+
+[ $UID -eq 0 ] && [ -e /etc/sysconfig/$prog ] && . /etc/sysconfig/$prog
+
+. /etc/init.d/functions
+
+if [ ! -e $config ] ; then
+    echo "No osqueryd config file found! at $config"
```
Comment (theopolis) [inline tools/deployment/osquery.initd]: We're currently installing an example config at /var/osquery/osquery.example.conf. Maybe we should change this location to /usr/share/osquery/osquery.example.conf and echo a note here that points to this /usr/share/osquery location for examples?

File: tools/deployment/osquery.initd [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+start() {
+    if [ $UID -ne 0 ] ; then
+        echo "User has insufficient privilege."
+        exit 4
+    elif [ -f $pidfile ]; then
+        PID=$(cat $pidfile)
+        echo "$prog is already running: $PID"
+        exit 4
+    else
+        daemon $prog --config_path=$config --daemonize=true
```
Comment (theopolis) [inline tools/deployment/osquery.initd]: Should *could* call:
```
daemon $prog --config_path=$config --pidfile=$pidfile --daemonize=true
```

Then we don't have to set pidfile in the config. Could do the same for lockfile too?

File: tools/deployment/osquery.initd [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+#              
+### END INIT INFO
+
+
+RETVAL=0
+prog="osqueryd"
+exec=/usr/bin/osqueryd
+config=/etc/osquery/osquery.conf
+lockfile=/var/lock/subsys/osqueryd
+pidfile=/var/run/osquery.pid
```
Comment (theopolis) [inline tools/deployment/osquery.initd]: @marpaia should we change the default pidfile location to `/var/run/osqueryd.pid`?

File: tools/deployment/osquery.initd (hunk 1 above)
Comment (maus-) [inline tools/deployment/osquery.initd]: That sounds good. I'll tweak it

File: tools/deployment/osquery.initd (hunk 2 above)
Comment (maus-) [inline tools/deployment/osquery.initd]: Yeah that'd be no problem.

**Comment by theopolis** [comment]
ok to test

File: tools/deployment/osquery.initd [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+if [ ! -e $config ] ; then
+    echo "No osqueryd config file found. at $config defaulting to example config"
+    if [ ! -e /usr/share/osquery/osquery.example.conf ] ; then
+        echo "No default config found"
+        exit 4
+    else
+        config=/usr/share/osquery/osquery.example.conf
+    fi
+fi
```
Comment (theopolis) [inline tools/deployment/osquery.initd]: How about this:
```
if [ ! -e $config ] ; then
    echo "No osquery config file found at $config."
    echo "See '/usr/share/osquery/osquery.example.conf' for an example config."
    exit 4
fi
```
then you get a chance to review the example config, it's options, and the example scheduled queries, before starting.

Comment (maus-) [inline tools/deployment/osquery.initd]: looks good to me!

**Pull request merged**